frontend/lib/util/get_application_url.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:ehrenamtskarte/build_config/build_config.dart';
+import 'package:ehrenamtskarte/configuration/definitions.dart';
+import 'package:ehrenamtskarte/constants/project_ids.dart';
+import 'package:ehrenamtskarte/identification/util/card_info_utils.dart';
+import 'package:ehrenamtskarte/proto/card.pb.dart';
+
+String getApplicationUrl(ApplicationUrl applicationUrl, bool isStagingEnabled) {
+  return isStagingEnabled
+      ? applicationUrl.staging
+      : isProduction()
+          ? applicationUrl.production
+          : applicationUrl.local;
+}
+
+String getApplicationUrlWithParameters(
+    String applicationUrl,
+    CardInfo cardInfo,
+    String projectId,
+    String? applicationUrlQueryKeyName,
+    String? applicationUrlQueryKeyBirthday,
+    String? applicationUrlQueryKeyReferenceNumber) {
```

Comment:
```suggestion
String getApplicationUrlWithParameters(BuildContext context, CardInfo cardInfo) {
  final applicationUrl = getApplicationUrl(context);
  final projectId = buildConfig.projectId.production;
  final applicationUrlQueryKeyName = buildConfig.applicationUrlQueryKeyName;
  final applicationUrlQueryKeyBirthday = buildConfig.applicationUrlQueryKeyBirthday;
  final applicationUrlQueryKeyReferenceNumber = buildConfig.applicationUrlQueryKeyReferenceNumber;
```

Reply:
if i change that, we need some linting ignores for null checks, since these values don't exist for all projects